test/e2e/apimachinery/certs.go

```diff
@@ -89,6 +88,12 @@ func setupServerCert(namespaceName, serviceName string) *certContext {
 	if err = ioutil.WriteFile(keyFile.Name(), privateKeyPEM, 0644); err != nil {
 		framework.Failf("Failed to write key file %v", err)
 	}
+	defer func() {
+		caCertFile.Close()
```

Comment:
Is this right?  Won't Failf abort the test, and potentially leave the files open if it fails before it gets here?

Reply:
You are right, thanks for reminding!